### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by hugopassos**

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    all_false = true
+    i = 0
+    while i < length
+      all_false = false if yield(self[i])
+      i += 1
+    end
+    all_false
+  end
+
+  def my_count
```
Comment (bellom) [inline main.rb]: Just one changes, and get a thumps up...
I got wrong output when i ran your code, error message below; 
image

**Review by bellom — CHANGES_REQUESTED** [review]
Hi @hugopassos,
Well done guy, kindly be aware that you are to use `Git Flow` but instead you used `Github`, make sure to use 'Git Flow' in the next project - `OOP`
Short Explanation Hint: 
`GitHub Flow`: Pull Requests is raise from feature branches to the master.
`Git Flow`: Pull Requests is raise from feature branches to the development.

You have done well, you only need to work on `my_count`, checkout my comment.

**Kindly re-submit for review after making the changes :slightly_smiling_face: Thumps -up await you...**

File: main.rb [hunk 2]
```diff
@@ -61,9 +61,9 @@ def my_count
...
     count = 0
-    i = 1
-    while i <= length
-      yield 1
+    i = 0
+    while i < length
+      yield self[i]
       count += 1
       i += 1
     end
```
Comment (maelfosso) [inline main.rb:68]: There is still a problem
```ruby
irb(main):002:0> ary = [1, 2, 4, 2]
=> [1, 2, 4, 2]
irb(main):003:0> ary.count
=> 4
irb(main):004:0> ary.my_count
=> 4
irb(main):005:0> ary.count(2)
=> 2
irb(main):006:0> ary.my_count(2)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):6
        1: from main.rb:60:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):007:0> ary.count{ |x| x%2 == 0 }
=> 3
irb(main):008:0> ary.my_count{ |x| x%2 == 0 }
=> 4
irb(main):009:0> 
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello,

Nice improvement. But there is still a problem.
Please, kindly read the comment left below and make the corresponding changes.
Once done, please re-submit the code review request.

Thanks

**Review by Ispirett — CHANGES_REQUESTED** [review]
Good Job!!! :+1:  you are almost there :+1:  

Your my_each_with_index method is broken please check against the original  ruby method 
make sure they outputting the same results thanks!!

**Review by Ispirett — APPROVED** [review]
Great job!!!! @hugopassos you can merge this project now :1st_place_medal:

**Pull request merged**